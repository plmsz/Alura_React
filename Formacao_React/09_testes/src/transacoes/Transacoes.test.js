import React from "react";
import { render } from "@testing-library/react";
import Transacoes from "./Transacoes";

xdescribe("Componente de transação do extrato", () => {
	it("O snapshot da lista de transacoes deve ser sempre o mesmo", () => {
		const listaDeTransacoes = [
			{ id: 1, tipo: "saque", valor: "10", data: "10/08/2020" },
			{ id: 2, tipo: "deposito", valor: "20", data: "26/09/2020" },
			{ id: 3, tipo: "deposito", valor: "100", data: "01/03/2022" },
		];

		const { container } = render(<Transacoes transacoes={listaDeTransacoes} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
