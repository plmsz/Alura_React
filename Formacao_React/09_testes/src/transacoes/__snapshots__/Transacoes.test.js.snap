// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato O snapshot da lista de transacoes deve ser sempre o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      10/08/2020
    </p>
    <p />
    <p>
      R$ 
      10
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      26/09/2020
    </p>
    <p />
    <p>
      R$ 
      20
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      01/03/2022
    </p>
    <p />
    <p>
      R$ 
      100
    </p>
  </div>
</div>
`;
